src/lib/utilities/history-summary-plucker.ts

```diff
@@ -0,0 +1,93 @@
+import { findValueIfItIncludesWord } from './find-value-if-it-includes-word';
+
+const summaryExtractors = {
+  ActivityTaskCancelRequested: (d) => ({ Id: d.activityId }),
+  ActivityTaskCompleted: (d) => ({ result: d.result }),
```

Comment:
For a bunch of these, we can probably have a utility method like `pick`.

```js
const pick = (source, ...keys) => keys.reduce((result, key) => {
  result[key] = source[key];
  return result;
}, {});
```

Or—if you prefer:

```js
const pick = (source, ...keys) => {
  const result = {};
  for (const key of keys) {
    result[key] = source[key];
  };
  return result;
}
```

Then, you could just do something like:

```js
const summaryExtractors = {
  // …
  ActivityTaskCompleted: (d) => pick(d, 'result'),
  ActivityTaskFailed: (d) => pick('details', 'reason'),
  // …
}
```

Basically, we want to avoid having bugs do to typos.

You could add some additional type safety even…

```ts
function pick<T, K extends keyof T>(source: T, ...keys: K[]): Pick<T, K> {
  const result: any = {};
  for (const key of keys) {
    result[key] = source[key];
  }
  return result;
}
```

This will stop us is from accidentally typing a key that does not exist on the source object.

There is probably even fancier stuff we could do with currying but the gymnastics we'd need to do with types probably isn't worth it.

Reply:
You can also do picks with destructuring like...

```typescript
  ActivityTaskFailed: ({details, reason}) => ({details, reason})
```

Although pick may be nicer.